wrapanapi/systems/openstack.py

```diff
@@ -824,6 +846,24 @@ def get_vm(self, name=None, id=None, ip=None):
             raise MultipleInstancesError('match criteria: {}'.format(kwargs))
         return matches[0]
 
+    def get_ports(self, uuid):
+        """
+        Get list of ports from openstack server
+
+        Args:
+            uuid (str): openstack server ID
+
+        Returns:
+            List of server ports
+
+        Example:
+            .. code-block:: python
+
+               mgmt.get_ports(uuid='server_uuid')
+        """
+        return [port['device_id'] for port in self.napi.list_ports()['ports']
+                if port['device_id'] == uuid]
```

Comment:
This method doesn't appear to be very useful - Its returning the UUID I just gave it, but in a list.

`get_ports` should take no arguments, and should return a list of all the ports.

I don't think `get_ports` method should take uuid, `find_port()` or `get_port` or `port_exists` should take uuid, and actually return something useful - not just the UUID that was passed. Return the actual openstack API port object when in doubt and let the caller do something with it.

Reply:
> This method doesn't appear to be very useful - Its returning the UUID I just gave it, but in a list.

My bad! @mshriver  I was suppose to return `port['id']` instead of `port['device_id']`.

> `get_ports` should take no arguments, and should return a list of all the ports.

Initially I thought not to use args for `get_ports` and rather use implementation something like `mgmt.get_vm("instance_name").get_ports` (utilizing nova vm object, like `volume_attached` property) but problem I saw was since we are accessing `OpenstackInstance`'s methods/properties I was not able to use `napi` there so I avoided that. May be I can just return all API port objects and let user parse whatever they want to do with it.

> I don't think `get_ports` method should take uuid, `find_port()` or `get_port` or `port_exists` should take uuid, and actually return something useful - not just the UUID that was passed. Return the actual openstack API port object when in doubt and let the caller do something with it.

Another reason I avoided that because openstack API port object consists of list of HUGE port info dictionaries  (something like https://paste.fedoraproject.org/paste/nNIEuasU2rEiigr10-x1Dw).

Let me know if you mean current changes.